src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlScriptRun.cs

```diff
@@ -67,19 +73,29 @@ public void PutFullDocument(
                 return;
 
             var commands = _fullDocuments[id] = new List<ICommandData>();
+            bool hasAttachments = attachments is { Count: > 0 };
+            if (hasAttachments)
+            {
+                DropRetiredAttachmentFlagFromAttachmentMetadataIfNeeded(context, id, doc, out doc);
```

Comment:
Shouldn't this always run?

Reply:
For that matter - IIRC, attachments are actually handled on the _receiving_ side, so if you don't send attachments, they'll be removed anyway

Reply:
Here I am actually stripping the Retired flag, and then unwrap (download) the attachment from cloud and send it in ETL